pkg/sloop/storemanager/storemanager.go

```diff
@@ -173,14 +173,53 @@ func (sm *StoreManager) refreshStats() *storeStats {
 	return sm.stats
 }
 
-func doCleanup(tables typed.Tables, timeLimit time.Duration, sizeLimitBytes int, stats *storeStats, deletionBatchSize int, gcThreshold float64) (bool, int64, int64, error) {
+func doCleanup(tables typed.Tables, timeLimit time.Duration, sizeLimitBytes int, stats *storeStats, deletionBatchSize int, gcThreshold float64, enableDeletePrefix bool) (bool, int64, int64, error) {
```

Comment:
I'm not asking as part of this PR, but hopefully one soon to follow

perhaps this is just me...
I'm struggling to understand the big/medium picture of this code (I think I did last set of changes as well). Under the umbrella of maintainability, is it possible to layer this code or abstract aspects of it so the flow of the logic is more easily observed.

Reply:
sure